packages/astro/src/content/runtime.ts

```diff
@@ -114,7 +103,7 @@ export function createGetEntryBySlug({
 				return render({
 					collection: entry.collection,
 					id: entry.id,
-					collectionToRenderEntryMap,
+					renderEntryImport: await getRenderEntryImport(collection, slug),
```

Comment:
```suggestion
					renderEntryImport: getRenderEntryImport(collection, slug),
```

Reply:
Ah, this `await` is needed actually! I got the type signature wrong 🙈